File: Note/app/src/main/java/com/zukron/note/model/ListNote.java
package com.zukron.note.model;

import java.util.ArrayList;

/**
 * Project name is Note
 * Created by Zukron Alviandy R on 8/8/2020
 */
public class ListNote {
    private Integer id;
    private String title;
    private ArrayList<Boolean> checks;
    private ArrayList<String> items;

    public ListNote(Integer id, String title, ArrayList<Boolean> checks, ArrayList<String> items) {
        this.id = id;
        this.title = title;
        this.checks = checks;
        this.items = items;
    }

    public Integer getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public ArrayList<Boolean> getChecks() {
        return checks;
    }

    public ArrayList<String> getItems() {
        return items;
    }
}

File: Note/app/src/main/java/com/zukron/note/model/ListNote.kt
package com.zukron.note.model

import androidx.room.*
import androidx.room.ForeignKey.CASCADE

/**
 * Project name is Note
 * Created by Zukron Alviandy R on 8/8/2020
 */
@Entity(
        tableName = "list_note",
        foreignKeys = [ForeignKey(
                entity = Note::class,
                parentColumns = arrayOf("id"),
                childColumns = arrayOf("note_id"),
                onUpdate = CASCADE,
                onDelete = CASCADE
        )],
        indices = [Index(value = ["note_id"], unique = false)]
)
data class ListNote(
        var check: Boolean,
        var item: String
) {
    @PrimaryKey(autoGenerate = true)
    var id: Long = 0
    var title: String = ""

    @ColumnInfo(name = "note_id")
    var noteId: Long = 0
}
